dotCMS/src/main/java/com/dotcms/rest/api/v1/template/TemplateResource.java

```diff
@@ -0,0 +1,752 @@
+     * @param response      {@link HttpServletResponse}
+     * @param templateForm  {@link TemplateForm}
+     * @return Response
+     * @throws DotDataException
+     * @throws DotSecurityException
+     */
+    @PUT
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    public final Response save(@Context final HttpServletRequest  request,
+                            @Context final HttpServletResponse response,
+                            final TemplateForm templateForm) throws DotDataException, DotSecurityException {
+
+        final InitDataObject initData = new WebResource.InitBuilder(webResource)
+                .requestAndResponse(request, response).rejectWhenNoUser(true).init();
+        final User user         = initData.getUser();
+        final Host host         = this.hostWebAPI.getCurrentHostNoThrow(request);
+        final PageMode pageMode = PageMode.get(request);
+        final Template currentTemplate = this.templateAPI.find(templateForm.getInode(), user, pageMode.respectAnonPerms);
+
+        if (null == currentTemplate || UtilMethods.isNotSet(currentTemplate.getIdentifier())
+                || !InodeUtils.isSet(currentTemplate.getInode())) {
+
+            throw new DoesNotExistException("The working template inode: " + templateForm.getInode() + " does not exists");
+        }
+
+        this.templateHelper.checkPermission(user, currentTemplate, PERMISSION_WRITE);
+
+        return Response.ok(new ResponseEntityView(this.templateHelper.toTemplateView(
+                this.fillAndSaveTemplate(templateForm, user, host, pageMode, currentTemplate), user))).build();
+    }
+
+    /**
+     * Publish a list of template inodes
+     * Return the list of a success published and failed
+     * @param request            {@link HttpServletRequest}
+     * @param response           {@link HttpServletResponse}
+     * @param templatesToPublish {@link List} list of template inodes to publish
+     * @return Response
+     * @throws DotDataException
+     * @throws DotSecurityException
+     */
+    @PUT
+    @Path("/_publish")
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    public final Response publish(@Context final HttpServletRequest  request,
+                               @Context final HttpServletResponse response,
+                               final List<String> templatesToPublish) throws DotDataException, DotSecurityException {
+
+        final InitDataObject initData = new WebResource.InitBuilder(webResource)
+                .requestAndResponse(request, response).rejectWhenNoUser(true).init();
+        final User user         = initData.getUser();
+        final Host host         = this.hostWebAPI.getCurrentHostNoThrow(request);
+        final PageMode pageMode = PageMode.get(request);
+        final List<String> publishedInodes = new ArrayList<>();
+        final List<String> failedInodes    = new ArrayList<>();
```

Comment:
Codacy found an issue: [Avoid declaring a variable if it is unreferenced before a possible exit point.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6042990)